### ISSUE TITLE
SA1513 and SA1500 for Auto-Implemented Properties

### ISSUE THREAD

**Issue by qmfrederik** [DotNetAnalyzers/StyleCopAnalyzers#1586]
SA1513 and SA1500 are triggered for [Auto-Implemented Properties](https://msdn.microsoft.com/en-us/library/bb384054.aspx)

For example, the following code:

```
public static DateTime Epoch
{
    get;
    private set;
} = new DateTime(1970, 1, 1);
```

would trigger

```
Warning	SA1513	Closing curly bracket must be followed by blank line
Warning	SA1500	Curly brackets for multi-line statements must not share line
```

The warning is technically correct (auto-implemented properties didn't exist at the time StyleCop was introduced), but it would seem odd to have StyleCop block a new C# feature.

**Comment by sharwell** [comment]
The warning is not triggered for either of the following:

```csharp
// Option 1
public static DateTime Epoch { get; } =
    new DateTime(1970, 1, 1);

// Option 2
public static DateTime Epoch { get; }
    = new DateTime(1970, 1, 1);
```

As of right now, no one has proposed a rule to enforce one way over the other. Each has their merits so I'm not sure which one I prefer either.

**Comment by sharwell** [comment]
I'm marking this resolved (answered), but if you're interested in a specific change in behavior please feel free to open a new issue describing the proposed change.

**Comment by qmfrederik** [comment]
I would personally prefer the `get;` and `set;` statements to each go on their own line; so my vote would go for 

```
public static DateTime Epoch
{
    get;
    private set;
} = new DateTime(1970, 1, 1);
```

being valid.

However, I most examples of auto property initializers did seem to favor the more condensed 
```
public static DateTime Epoch { get; } = new DateTime(1970, 1, 1);
```

which is probably what I'll settle on for now.

Thanks!

**Comment by sharwell** [comment]
Removing the setter does seem to solve all the problems. :smile:
